webapplication/aoi/urls.py

```diff
@@ -12,6 +12,8 @@
     
     path('notebook', ComponentListCreateAPIView.as_view(), name='notebook_list_or_create'),
     path('notebook/<int:pk>', ComponentRetrieveUpdateDestroyAPIView.as_view(), name='notebook'),
+    path('notebook/<int:pk>/price/<int:aoi>', ComponentPriceAPIView.as_view(),
+         name='calculate_notebook_execution_price'),
```

Comment:
This doesn't fit into the REST API. Lets do the following. When POSTing to the 'request' (RequestListCreateAPIView) add boolean attribute presubmit. By default it should be false or None. If presubmit is True then do not store Request (and Transaction) and do not check balance, but return resulting Request object (with price). In this way we can calculate price not only by area but also by other model parameters passed to Request (if needed) and return not only price but other details of the resulting Request object (if needed)